contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -138,43 +147,46 @@ contract NodeOperatorsRegistry is INodeOperatorsRegistry, IsContract, AragonApp
     }
 
     /**
-      * @notice `_active ? 'Enable' : 'Disable'` the node operator #`_id`
-      */
-    function setNodeOperatorActive(uint256 _id, bool _active) external
+     * @notice `_active ? 'Enable' : 'Disable'` the node operator #`_id`
+     */
+    function setNodeOperatorActive(uint256 _id, bool _active)
+        external
         authP(SET_NODE_OPERATOR_ACTIVE_ROLE, arr(_id, _active ? uint256(1) : uint256(0)))
         operatorExists(_id)
     {
         require(operators[_id].active != _active, "NODE_OPERATOR_ACTIVITY_ALREADY_SET");
 
         _increaseKeysOpIndex();
 
+        operators[_id].active = _active;
+
         uint256 activeOperatorsCount = getActiveNodeOperatorsCount();
-        if (_active)
+        if (_active) {
             ACTIVE_OPERATORS_COUNT_POSITION.setStorageUint256(activeOperatorsCount.add(1));
-        else
+        } else {
             ACTIVE_OPERATORS_COUNT_POSITION.setStorageUint256(activeOperatorsCount.sub(1));
 
-        operators[_id].active = _active;
+            _trimUnusedNodeOperatorKeys(_id);
+            _updateTotalAvailableKeysCount();
+        }
 
         emit NodeOperatorActiveSet(_id, _active);
     }
 
     /**
-      * @notice Change human-readable name of the node operator #`_id` to `_name`
-      */
-    function setNodeOperatorName(uint256 _id, string _name) external
-        authP(SET_NODE_OPERATOR_NAME_ROLE, arr(_id))
-        operatorExists(_id)
-    {
+     * @notice Change human-readable name of the node operator #`_id` to `_name`
+     */
+    function setNodeOperatorName(uint256 _id, string _name) external authP(SET_NODE_OPERATOR_NAME_ROLE, arr(_id)) operatorExists(_id) {
         require(keccak256(operators[_id].name) != keccak256(_name), "NODE_OPERATOR_NAME_IS_THE_SAME");
```

Comment:
maybe restrict `_name` at least by length?
something like >0 and <255

Reply:
fixed